bot/seasons/evergreen/wikipedia.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from typing import Optional
+
+from discord.ext import commands
+
+log = logging.getLogger(__name__)
+
+SEARCH_API = "https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch={search_term}&format=json"
+WIKIPEDIA_URL = "https://en.wikipedia.org/wiki/{title}"
+
+
+class WikipediaCog(commands.Cog):
+    """Get info from wikipedia."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    async def search_wikipedia(self, search_term: str) -> Optional[str]:
+        """Search wikipedia and return the first page found."""
+        async with self.http_session.get(SEARCH_API.format(search_term=search_term)) as response:
+            data = await response.json()
+
+        search_result = data["query"]["search"]
+        if len(search_result) == 0:
+            return None
+
+        page = search_result[0]
+
+        # we dont like "may refere to" pages.
+        if "may refer to" in page["snippet"]:
+            log.info(f"hit a 'may refer to' page, taking result 2 (index 1)")
+            page = search_result[1]
```

Comment:
We can definitely use some logic here instead of relying on the very first result, specially when they are sorted alphabetically and not by how close the match is, for example when searching for `Avenger` I got this:

![image](https://user-images.githubusercontent.com/19921765/78028629-835f5c00-7389-11ea-95f9-ad9e6ff9d836.png)

And the first hit is this:
![image](https://user-images.githubusercontent.com/19921765/78028661-8bb79700-7389-11ea-8261-5f3795376864.png)

I mean, when searching wikipedia for Marvel's Avenger, I definitely am not searching for a `Deer Avenger` right? :D